File: maps/src/main/java/it/localhostsoftware/maps/huawei/model/HuaweiLatLng.java
package it.localhostsoftware.maps.huawei.model;

import it.localhostsoftware.maps.model.LatLng;

public class HuaweiLatLng extends LatLng<com.huawei.hms.maps.model.LatLng> {
    public HuaweiLatLng(com.huawei.hms.maps.model.LatLng latLng) {
        super(latLng);
    }

    @Override
    public double getLatitude() {
        return getLatLng().latitude;
    }

    @Override
    public double getLongitude() {
        return getLatLng().longitude;
    }
}

File: maps/src/main/kotlin/it/localhostsoftware/maps/huawei/model/HuaweiLatLng.kt
package it.localhostsoftware.maps.huawei.model

import com.huawei.hms.maps.model.LatLng

class HuaweiLatLng(ll: LatLng) : it.localhostsoftware.maps.model.LatLng<LatLng>(ll) {
    override val latitude: Double
        get() = ll.latitude
    override val longitude: Double
        get() = ll.longitude
}